src/easier68k/assembler/assembler.py

```diff
@@ -37,10 +46,93 @@ def parse(text: str):  # should return a list file and errors/warnings eventuall
+
+        # Replace all substitutions in the current line with their corresponding values
+        for equate in equates.items():
+            contents = contents.replace(equate[0], equate[1])
+
+        if opcode == 'ORG':  # This will shift our current memory location, it's a special case
+            parsed = parse_literal(contents)
+            # TODO: Max/min memory location?
+            current_memory_location = int.from_bytes(parsed, 'big')
+            continue
+
+        if has_label(stripped):
+            label = get_label(stripped)
+            label_addresses[label] = current_memory_location
+            to_return.define_symbol(label, current_memory_location)
+
+        # We don't know this opcode, there's no module for it
+        if opcode.lower() not in opcodes.__all__:
+            issues.append(('Opcode {} not known, but continuing and dropping it.', 'ERROR'))
+            continue
+
+        op_module = sys.modules['easier68k.core.opcodes.{}'.format(opcode.lower())]
+        op_class = getattr(op_module, opcode.capitalize())
+        length, issues = op_class.get_word_length(opcode, contents)
+
+        current_memory_location += length * 2
+
+    # --- PART 3: actually create the list file ---
+    for line_index, stripped in for_line_stripped_comments(text):
+        opcode = get_opcode(stripped).upper()
+
+        # Equates have already been processed, skip them
+        # (this idea could be expanded for more preprocessor directives)
+        if opcode == 'EQU':
+            continue
+
+        contents = strip_opcode(stripped)
+
+        # Replace all substitutions in the current line with their corresponding values
+        for equate in equates.items():
+            contents = contents.replace(equate[0], equate[1])
+
+        # Replace all memory labels with their proper values (that's the difference in this step)
+        for label in label_addresses.items():
+            contents = contents.replace(label[0], '$' + hex(label[1])[2:])
+
+        if opcode == 'ORG':  # This will shift our current memory location, it's a special case
+            parsed = parse_literal(contents)
+            new_memory_location = int.from_bytes(parsed, 'big')
+            assert 0 <= new_memory_location < 16777216, 'ORG address must be between 0 and 2^24!'
+            current_memory_location = new_memory_location
+            continue
+
+        # We don't know this opcode, there's no module for it
+        if opcode.lower() not in opcodes.__all__:
+            issues.append(('Opcode {} not known, but continuing and dropping it.', 'ERROR'))
+            continue
+
+        # Get the module of this opcode
+        op_module = sys.modules['easier68k.core.opcodes.{}'.format(opcode.lower())]
+
+        # Get the class of this opcode from inside the module
+        op_class = getattr(op_module, opcode.capitalize())
+
+        # Get the actual constructed opcode
+        data, issues = op_class.from_str(opcode, contents)
```

Comment:
same thing from before.

also, may want to have code reuse for this

Reply:
Not sure which same thing you're referring to, but I'm taking my best guess at it in an upcoming commit.

Reply:
Also, I don't think code reuse is too necessary here unless you're thinking you'll have similar functionality in the simulator/disassembler. I think I'll go through the assembler in general and try to shorten the code by splitting some things out though.

Reply:
I take it back: I forgot I have basically duped code in here, I'm going through and doing reuse now.